RFC-0004-expression-eval.md

```diff
@@ -0,0 +1,80 @@
+# **RFC0 for Presto**
+
+## [Title]
+
+Proposers
+
+* Tim Meehan
+
+## [Related Issues]
+
+* RFC-0003
+
+## Summary
+
+Constant folding is a technique used in compilers to evaluate constant expressions at compile time. This RFC proposes to add constant folding to Presto's expression evaluation engine. This will allow Presto to evaluate constant expressions at query planning time, reducing the amount of work that needs to be done at query execution time.
```

Comment:
> This RFC proposes to add constant folding to Presto's expression evaluation engine.

Doesn't Presto already have constant folding?

Reply:
Yes, I meant something along the lines of the following: "This RFC proposes to add pluggable constant folding support to Presto's optimizer so that the underlying evaluation engine can simplify expressions."  I'll update the doc, and concisely mention the motivation here as well.